core/components/product-card/delete-wishlist-item-form/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { Heart } from 'lucide-react';
+import { useTranslations } from 'next-intl';
+import { toast } from 'react-hot-toast';
+
+import { Button } from '~/components/ui/button';
+
+import { deleteWishlistItem } from './_actions/delete-wishlist-item';
+
+const SubmitButton = () => {
+  const t = useTranslations('Components.ProductCard.DeleteWishlistItem');
+
+  return (
+    <Button
+      aria-label={t('deleteAriaLabel')}
+      className="p-3 text-black hover:bg-transparent hover:text-black"
+      title={t('delete')}
+      type="submit"
+      variant="subtle"
+    >
+      <Heart fill="currentColor" />
+    </Button>
+  );
+};
+
+export interface DeleteWishlistItemFormProps {
+  wishlistId: number;
+  wishlistItemId: number;
+  onWishlistItemDelete: (itemEntityId: number) => void;
+  onWishlistItemDeleted: (itemEntityId: number) => void;
+  onWishlistItemUndoDelete: (itemEntityId: number) => void;
+}
+
+export const DeleteWishlistItemForm = ({
+  onWishlistItemDelete,
+  onWishlistItemDeleted,
+  onWishlistItemUndoDelete,
+  wishlistId,
+  wishlistItemId,
+}: DeleteWishlistItemFormProps) => {
+  const t = useTranslations('Components.ProductCard.DeleteWishlistItem');
+
+  return (
+    <form
+      action={(formData: FormData) => {
+        const entityId = Number(formData.get('wishlistId'));
+        const itemEntityId = Number(formData.get('wishlistItemId'));
+
+        onWishlistItemDelete(itemEntityId);
+
+        let toastId: string;
+
+        const timer = setTimeout(() => {
+          toast.dismiss(toastId);
+
+          onWishlistItemDeleted(itemEntityId);
+
+          void deleteWishlistItem({
+            input: { entityId, itemEntityIds: [itemEntityId] },
+          });
+        }, 5000);
```

Comment:
This is workaround for current [react-hot-toast](https://www.npmjs.com/package/react-hot-toast) library we use. In future we may change it to [react-toastify](https://www.npmjs.com/package/react-toastify) that has useful onClose callback needed to pass deleteWishlistItem mutation in